Libraries/LibJS/AST.cpp

```diff
@@ -1198,84 +1198,67 @@ void ThisExpression::dump(int indent) const
 
 Value AssignmentExpression::execute(Interpreter& interpreter, GlobalObject& global_object) const
 {
-    auto rhs_result = m_rhs->execute(interpreter, global_object);
-    if (interpreter.exception())
-        return {};
+#define EXECUTE_LHS_AND_RHS()                                    \
+    do {                                                         \
+        lhs_result = m_lhs->execute(interpreter, global_object); \
+        if (interpreter.exception())                             \
+            return {};                                           \
+        rhs_result = m_rhs->execute(interpreter, global_object); \
+        if (interpreter.exception())                             \
+            return {};                                           \
+    } while (0)
```

Comment:
I tried doing this with a two value struct the (like `compute_this_and_callee()`) but couldn't get it working in a concise way because of the `auto` in `auto [lhs_result, rhs_result] = ...`...

Reply:
Perhaps we should have something akin to `std::bind`?
Though that particular function is...ugly imo